collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -41,32 +42,43 @@
 @SuppressLint("ViewConstructor")
 public class UrlWidget extends QuestionWidget implements ButtonWidget {
 
-    private Uri uri;
-    final Button openUrlButton;
-    final TextView stringAnswer;
     private final CustomTabHelper customTabHelper;
+    private Uri uri;
 
-    public UrlWidget(Context context, QuestionDetails questionDetails) {
+    protected Button openUrlButton;
+    protected TextView stringAnswer;
+
+    public UrlWidget(Context context, QuestionDetails questionDetails, CustomTabHelper customTabHelper) {
         super(context, questionDetails);
+        this.customTabHelper = customTabHelper;
+    }
 
-        openUrlButton = createSimpleButton(getContext(), getFormEntryPrompt().isReadOnly(), context.getString(R.string.open_url), getAnswerFontSize(), this);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        ViewGroup answerView = (ViewGroup) LayoutInflater.from(context).inflate(R.layout.url_widget_answer, null);
 
-        stringAnswer = getCenteredAnswerTextView(getContext(), getAnswerFontSize());
+        openUrlButton = answerView.findViewById(R.id.url_button);
+        stringAnswer = answerView.findViewById(R.id.url_answer_text);
 
-        String s = questionDetails.getPrompt().getAnswerText();
+        if (prompt.isReadOnly()) {
+            openUrlButton.setVisibility(GONE);
+        } else {
+            openUrlButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+            openUrlButton.setOnClickListener(v -> {
+                if (MultiClickGuard.allowClick(QuestionWidget.class.getName())) {
```

Comment:
I would move this check to onButtonClick() then this expression will look simpler.